code/datums/components/chameleon/_chameleon.dm

```diff
@@ -0,0 +1,179 @@
+		return
+	if(!(disguise_path in disguise_paths))
+		if(user && ispath(disguise_path, /obj/item))
+			var/obj/item/failed_disguise = disguise_path
+			to_chat(user, "<span class='warning'>The Chameleon [original_name] cannot disguise as '[initial(failed_disguise.name)] ([initial(failed_disguise.icon_state)])'.</span>")
+		return
+	var/old_disguise_path = current_disguise
+	var/obj/item/picked_item = disguise_path
+	var/obj/item/chameleon_item = parent
+	chameleon_item.worn_icon = initial(picked_item.worn_icon)
+	chameleon_item.lefthand_file = initial(picked_item.lefthand_file)
+	chameleon_item.righthand_file = initial(picked_item.righthand_file)
+	if(initial(picked_item.greyscale_colors))
+		chameleon_item.greyscale_colors = initial(picked_item.greyscale_colors)
+		if(initial(picked_item.greyscale_config_worn))
+			chameleon_item.worn_icon = SSgreyscale.GetColoredIconByType(initial(picked_item.greyscale_config_worn), initial(picked_item.greyscale_colors))
+		if(initial(picked_item.greyscale_config_inhand_left))
+			chameleon_item.lefthand_file = SSgreyscale.GetColoredIconByType(initial(picked_item.greyscale_config_inhand_left), initial(picked_item.greyscale_colors))
+		if(initial(picked_item.greyscale_config_inhand_right))
+			chameleon_item.righthand_file = SSgreyscale.GetColoredIconByType(initial(picked_item.greyscale_config_inhand_right), initial(picked_item.greyscale_colors))
+	chameleon_item.worn_icon_state = initial(picked_item.worn_icon_state)
+	chameleon_item.item_state = initial(picked_item.item_state)
+	if(isclothing(chameleon_item) && ispath(picked_item, /obj/item/clothing))
+		var/obj/item/clothing/chameleon_clothing = chameleon_item
+		var/obj/item/clothing/picked_clothing = picked_item
+		chameleon_clothing.flags_cover = initial(picked_clothing.flags_cover)
+	if(initial(picked_item.greyscale_config) && initial(picked_item.greyscale_colors))
+		chameleon_item.icon = SSgreyscale.GetColoredIconByType(initial(picked_item.greyscale_config), initial(picked_item.greyscale_colors))
+	else
+		chameleon_item.icon = initial(picked_item.icon)
+	chameleon_item.desc = initial(picked_item.desc)
+	chameleon_item.icon_state = initial(picked_item.icon_state)
+	if(istype(chameleon_item, /obj/item/card/id) && ispath(disguise_path, /obj/item/card/id))
+		var/obj/item/card/id/picked_id = disguise_path
+		var/obj/item/card/id/chameleon_id = chameleon_item
+		chameleon_id.hud_state = initial(picked_id.hud_state)
+	else
+		chameleon_item.name = initial(picked_item.name)
+	chameleon_item.update_slot_icon()
+	current_disguise = disguise_path
+	on_disguise?.Invoke(user, src, old_disguise_path, disguise_path)
+
+/datum/component/chameleon/proc/setup_disguises()
+	disguise_paths.Cut()
+	for(var/path in list_chameleon_disguises(base_disguise_path, disguise_blacklist, hide_duplicates))
+		disguise_paths += path
+		GLOB.all_available_chameleon_types |= path
```

Comment:
confused by this glob here. why is this global picking paths from a random proc

Reply:
what do you mean?

Reply:
oh i see. i uh, forgot what this glob was for lol.